taichi/python/export_lang.cpp

```diff
@@ -103,6 +103,7 @@ void export_lang(py::module &m) {
       .def(py::self == py::self)
       .def("__hash__", &DataType::hash)
       .def("to_string", &DataType::to_string)
+      .def("__str__", &DataType::to_string)
```

Comment:
do we need add `__repr__` too?

Reply:
SGTM!

Reply:
`__repr__` is more like for debugging purpose which object information. 
I usually think in this way (maybe incorrect :P):
information from `__str__` should be human readable (but can lose info)
But you can use information from `__repr__` to reconstruct this object easily. 
So IMHO only handling `__str__` is good enough for now and we might consider adding `__repr__` in the future :D